src/ripple/rpc/impl/ServerHandlerImp.cpp

```diff
@@ -377,6 +377,34 @@ ServerHandlerImp::onStopped(Server&)
 
 //------------------------------------------------------------------------------
 
+template <class T>
+void
+logDuration(
+    Json::Value const& request,
+    T const& duration,
+    beast::Journal& journal)
+{
+    std::stringstream logMsg;
+    logMsg << "RPC request processing duration = "
+           << std::chrono::duration_cast<std::chrono::microseconds>(duration)
+                  .count()
+           << " microseconds. request = " << request;
+    if (std::chrono::duration_cast<std::chrono::seconds>(duration).count() >=
+        10)
+    {
+        JLOG(journal.error()) << logMsg.str();
+    }
+    else if (
+        std::chrono::duration_cast<std::chrono::seconds>(duration).count() >= 1)
+    {
+        JLOG(journal.warn()) << logMsg.str();
+    }
+    else
+    {
+        JLOG(journal.debug()) << logMsg.str();
+    }
+}
+
```

Comment:
Consider trying this. This is off the top of the dome, so it might require some finessing.

```cpp
auto const seconds = std::chrono::duration_cast<std::chrono::seconds>(duration).count();
using J = decltype(journal);
auto const level = (seconds >= 10) ? &J::error : (seconds >= 1) ? &J:warn : &J:debug;
JLOG(journal.*level()) << "RPC request processing duration = "
           << std::chrono::duration_cast<std::chrono::microseconds>(duration)
                  .count()
           << " microseconds. request = " << request;
```

Reply:
```
    auto const seconds =
        std::chrono::duration_cast<std::chrono::seconds>(duration).count();

    auto const level = (seconds >= 10)
        ? journal.error()
        : (seconds >= 1) ? journal.warn() : journal.debug();

    JLOG(level) << "RPC request processing duration = "
                << std::chrono::duration_cast<std::chrono::microseconds>(
                       duration)
                       .count()
                << " microseconds. request = " << request;
```

worked for me. Tested by logging a request that ran for > 1s, and got the appropriate warn() message.